Mage.Sets/src/mage/cards/c/ConsumingFerocity.java

```diff
@@ -82,16 +86,15 @@ public ConsumingFerocityEffect copy() {
 
     @Override
     public boolean apply(Game game, Ability source) {
-        Permanent creature = game.getPermanent(source.getFirstTarget());
-        if (creature != null) {
-            Effect effect = new AddCountersAttachedEffect(CounterType.P1P0.createInstance(), "enchanted creature");
-            effect.apply(game, source);
+        Permanent permanent = game.getPermanentOrLKIBattlefield(source.getSourceId());
+        if (permanent != null && permanent.getAttachedTo() != null) {
+            Permanent creature = game.getPermanent(permanent.getAttachedTo());
```

Comment:
Don't need to null check the UUID `permanent.getAttachedTo()` as `game.getPermanent()` can handle null values, but do need to null check the resulting `Permanent creature` as it is accessed below